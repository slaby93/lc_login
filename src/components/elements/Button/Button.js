import styled, { css } from 'styled-components';
import { prop, ifProp } from 'styled-tools';

const StyledComponent = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid ${prop('theme.color.black')};
  font-size: 16px;
  background-color: ${prop('theme.color.white')};
  border-radius: 2px;
  border-color: ${prop('theme.color.dirtyWhite')};
  padding: 10px;
  color:  ${prop('color', prop('theme.color.black'))};
  background-color:  ${prop('bgColor', prop('theme.color.white'))};
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px ${prop('theme.color.whiteShadow')};
  transition: all .5s ease-out, opacity .5s ease-out;
  outline: none;
  cursor: pointer;
  &:active {
    box-shadow: 2px 2px 7px 0px ${prop('theme.color.dirtyShadow')};
  }
  &:hover:not(:active) {
    transform: scale(1.03);
  }
  ${ifProp('grow', css`
      width: 100%;
  `)}
  ${ifProp('disabled', css`
     cursor: not-allowed;
     opacity: 0.4;
  `)}
  ${ifProp('round', css`
     border-radius: 50%;
  `)}
  ${ifProp('text', css`
      display: inline;
      border: none;
      box-shadow: none;
      margin-bottom: 10px;
      height: initial;
      margin: 0;
      width: fit-content;
      &:active {
        box-shadow:none;
      }
      &:hover {
        transform: scale(1.05);
      }
  `)}

  ${ifProp('noBorder', css`
    border: none;
  `)}
`;

export default StyledComponent;
