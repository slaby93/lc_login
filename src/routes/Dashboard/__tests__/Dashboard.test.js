import React from 'react';
import { shallow } from 'enzyme';
import Button from 'components/elements/Button';
import DashboardStyled, { Dashboard } from './../Dashboard';

describe('<Dashboard />', () => {
  test('CSS', () => {
    const mountedComponent = shallow(<DashboardStyled
      onLogout={jest.fn()}
    />);
    expect(mountedComponent).toMatchSnapshot();
  });

  test('HTML', () => {
    const mountedComponent = shallow(<Dashboard
      onLogout={jest.fn()}
      className="test"
    />);
    expect(mountedComponent).toMatchSnapshot();
  });

  test('Button click', () => {
    const onLogoutMock = jest.fn();
    const mountedComponent = shallow(<Dashboard
      onLogout={onLogoutMock}
      className="test"
    />);
    expect(onLogoutMock).not.toBeCalled();
    mountedComponent.find(Button).simulate('click');
    expect(onLogoutMock).toBeCalled();
  });
});
