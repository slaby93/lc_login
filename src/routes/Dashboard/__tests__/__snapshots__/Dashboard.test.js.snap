// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> CSS 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .Button__StyledComponent-hhslSX {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-top: 15%;
}

<Dashboard
  className="c0"
  onLogout={[Function]}
/>
`;

exports[`<Dashboard /> HTML 1`] = `
<div
  className="test"
>
  <Button__StyledComponent
    onClick={[Function]}
  >
    Logout
  </Button__StyledComponent>
</div>
`;
